samples/createSubscriptionWithFiltering.js

```diff
@@ -0,0 +1,78 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * This application demonstrates how to perform basic operations on
+ * subscriptions with the Google Cloud Pub/Sub API.
+ *
+ * For more information, see the README.md under /pubsub and the documentation
+ * at https://cloud.google.com/pubsub/docs.
+ */
+
+// This is a generated sample. Please see typescript/README.md for more info.
+
+'use strict';
+
+// sample-metadata:
+//   title: Create Subscription With Filtering
+//   description: Creates a new subscription with filtering.
+//   usage: node createSubscriptionWithFiltering.js <topic-name-or-id> <subscription-name-or-id> <filter-string>
+
+// [START pubsub_create_subscription_with_filter]
+/**
+ * TODO(developer): Uncomment these variables before running the sample.
+ */
+// const topicNameOrId = 'YOUR_TOPIC_NAME_OR_ID';
+// const subscriptionNameOrId = 'YOUR_SUBSCRIPTION_NAME_OR_ID';
+// const filterString = 'YOUR_FILTER_STRING';
```

Comment:
Should we give an example here like we do in the [Java samples](https://cloud.google.com/pubsub/docs/samples/pubsub-create-subscription-with-filter#pubsub_create_subscription_with_filter-java)?

Reply:
Yeah, it's unclear to me since I don't want to give the impression, compared to the other default strings, that there's a "correct" filter to put there. I might add the sample filter as a comment after the rest of it...?